packages/terra-framework-site/src/app/appConfig.jsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import Home from './home/Home';
+import ApplicationMenu from './ApplicationMenu';
+import Tests from './tests/Tests';
+import TestsMenu from './tests/TestsMenu';
+
+import componentConfig from '../componentConfig';
+import Components from './components/Components';
+import ComponentsMenu from './components/ComponentsMenu';
+import ComponentTests from './tests/ComponentTests';
+import ComponentTestsMenu from './tests/ComponentTestsMenu';
+
+const injectConfig = config => (
+  ComponentClass => (
+    props => (
+      <ComponentClass {...props} config={config} />
+    )
+  )
+);
+
+const componentTests = {};
+Object.keys(componentConfig).map((componentKey) => {
+  const testRoot = componentConfig[componentKey].testRoot;
+  if (testRoot) {
+    return {
+      path: testRoot,
+      component: {
+        default: {
+          componentClass: injectConfig(componentConfig[componentKey])(ComponentTests),
+        },
+      },
+    };
+  }
+  return undefined;
+})
+.filter(test => !!test)
+.forEach((test) => {
+  componentTests[test.path] = test;
+});
+
+const componentTestMenus = {};
+Object.keys(componentConfig).map((componentKey) => {
+  const testRoot = componentConfig[componentKey].testRoot;
+  if (testRoot) {
+    return {
+      path: testRoot,
+      component: {
+        default: {
+          componentClass: injectConfig(componentConfig[componentKey])(ComponentTestsMenu),
+        },
+      },
+    };
+  }
+  return undefined;
+})
+.filter(test => !!test)
```

Comment:
test => !!test?  what is this doing?

Reply:
'!!test' will cast 'test' to a boolean. The filter function uses that to remove undefined values from the array that the previous .map call produced.

Basically it's Ruby's compact function: http://ruby-doc.org/core-1.9.3/Array.html#method-i-compact

Reply:
I had to look this up too 😄